### PR TITLE
[spirv] Support i8 matmul codegen with integer dot prod ops

### PR DESCRIPTION
**Pull request by kuhar**
This configuration is enabled conditionally for targets that support the necessary extension and capabilities.

### CODE REVIEW

**Comment by kuhar** [comment]
I undid the tiling size changes -- some upstream lowering changed and this requires more massaging to work with bigger K0.  I need to look into it.

Edit: Done.

File: compiler/src/iree/compiler/Codegen/SPIRV/SPIRVVectorize.cpp [hunk 1]
```diff
@@ -81,17 +86,108 @@ SmallVector<int64_t> getNativeVectorShapeImpl(VectorTransferOpInterface op) {
...
+/// Succeeds when |contract| is a i32 matmul whose LHS and RHS operands may be
+/// result of zero/sign extension of i8 inputs.
+LogicalResult detectI8ToI32Matmul(vector::ContractionOp contract) {
+  if (contract.getKind() != vector::CombiningKind::ADD) return failure();
+
+  if (!mayExtI8ToI32(contract.getLhs()) || !mayExtI8ToI32(contract.getRhs()))
+    return failure();
+
+  AffineExpr m, n, k;
+  bindDims(contract.getContext(), m, n, k);
```
Comment (antiagainst) [inline compiler/src/iree/compiler/Codegen/SPIRV/SPIRVVectorize.cpp]: These bound dims aren't used?

**Review by antiagainst — APPROVED** [review]
Awesome!!

File: compiler/src/iree/compiler/Codegen/SPIRV/SPIRVVectorize.cpp (hunk 1 above)
Comment (kuhar) [reply compiler/src/iree/compiler/Codegen/SPIRV/SPIRVVectorize.cpp]: thanks!

**Pull request merged**